src/main/scala/mesosphere/marathon/Main.scala

```diff
@@ -52,29 +53,64 @@ class MarathonApp(args: Seq[String]) extends AutoCloseable with StrictLogging {
   val config: Config = {
     // eventually we will need a more robust way of going from Scallop -> Config.
     val overrides = {
+
+      //
+      // Create Kamon configuration spec for the `kamon.datadog` plugin
+      //
       val datadog = cliConf.dataDog.get.map { urlStr =>
-        val url = new URI(urlStr)
+        val url = Uri(urlStr)
+        val params = url.query()
+
         s"""
-      |kamon.datadog {
-      |hostname: ${url.getHost}
-      |port: ${if (url.getPort == -1) 8125 else url.getPort}
-      |}
-        """.stripMargin
+           |kamon.datadog {
+           |  hostname: ${url.authority.host}
+           |  port: ${if (url.authority.port == 0) 8125 else url.authority.port}
+           |  flush-interval: ${params.collectFirst { case ("interval", iv) => iv.toInt }.getOrElse(10)} seconds
+           |""".stripMargin +
+          params.foldLeft("")(
+            (expr: String, param) => param._1 match {
+              case "prefix" => expr + s"  application-name: ${param._2}\n"
```

Comment:
Seems like it might help to use `ConfigFactory.parseMap` here, too?

Reply:
Yup, I like the approach. Working on it.